bridge-fungible-token/src/bridge_fungible_token.sw

```diff
@@ -71,31 +79,47 @@ impl MessageReceiver for Contract {
         };
 
         let res_amount = adjust_deposit_decimals(message_data.amount, DECIMALS, BRIDGED_TOKEN_DECIMALS);
+
         match res_amount {
             Result::Err(e) => {
                 // register a refund if value can't be adjusted
                 register_refund(message_data.from, message_data.token, message_data.amount);
             },
-            Result::Ok(a) => {
-                storage.tokens_minted += a;
-                mint_to_address(a, message_data.to);
+            Result::Ok(amount) => {
+                match storage.tokens_minted.try_read() {
+                    Option::Some(value) => storage.tokens_minted.write(value + amount),
+                    Option::None => storage.tokens_minted.write(amount),
+                };
+                mint_to(amount, message_data.to);
                 log(DepositEvent {
                     to: message_data.to,
                     from: message_data.from,
-                    amount: a,
+                    amount: amount,
                 });
+
+                // when depositing to an address, the message is 160 bytes. If it is greater,
+                // we know that we need to call `process_message()` on the receiving contract.
+                if message_data.len > 161 {
+                    match message_data.to {
+                        Identity::ContractId(id) => {
+                            let dest_contract = abi(MessageReceiver, id.into());
+                            dest_contract.process_message(msg_idx);
```

Comment:
We should be pushing the coins in this contract call, so the contract can verify the coins it's receiving

Reply:
Yeah, that needs a refactor for sure. This [issue](https://github.com/FuelLabs/bridge-fungible-token/issues/61) is for adding testing for this workflow which will confirm it's working correctly; So far, only the actual deposit to the contract is verified.